### PR TITLE
JIRA: ABC-285 Material faceset name order fix.

### PR DESCRIPTION
**Pull request by clusty**
This is a similar bug to the old: "sub mesh order is not the same as the FBX one".
In this particular case, the wrong index was being applied when sorting the face set names.

JIRA: https://jira.unity3d.com/browse/ABC-285

@cguer I'd say it safe to test the original steps.
If you can export an abc without faceset names, check that it does not crash.

NOTE: Will add a test as soon as I have an asset that repro's the problem.

### CODE REVIEW

**Comment by unity-cla-assistant** [comment]
[CLA assistant check](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.formats.alembic?pullRequest=496) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.cds.internal.unity3d.com/Unity-Technologies/com.unity.formats.alembic?pullRequest=496) before we can accept your contribution.<br/><hr/>**noreply@unity3d.com** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.cds.internal.unity3d.com/check/Unity-Technologies/com.unity.formats.alembic?pullRequest=496) it.</sub>

File: com.unity.formats.alembic/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 ### Changed
 ### Fixed
+- The faceset names were being read in the incorect order in certain cases.
```
Comment (sebastienduverne) [inline com.unity.formats.alembic/CHANGELOG.md:11]: Suggestion:
`Make sure Unity always reads the Face Set names in the correct order.`

**Review by cguer — APPROVED** [review]
QA approved 👍!

Tested on:
- 2021.2.7f1 (Windows only)

**Review by mfe — APPROVED** [review]
LGTM, though I couldn't fully repro with the fly car test package because of an error on one of the asset : 
`Prefab instance problem: fly_car (Missing Prefab)`

Still, code is pretty straightforward and Christophe tested. So all good.

**Review by sebastienduverne — APPROVED** [review]
Changelog ok

**Pull request merged**